src/vg_client_pool.erl

```diff
@@ -1,22 +1,119 @@
 -module(vg_client_pool).
 
 -export([start/0,
-         start/1,
-         stop/0]).
+         stop/0,
+         get_pool/2,
+         refresh_topic_map/0]).
+
+-define(topic_map, topic_map).
+
+-define(OPTIONS, [set, public, named_table, {read_concurrency, true}]).
+
+-record(chain, {
+          name :: binary(),
+          head_host :: inet:ip_address() | inet:hostname(),
+          head_port :: inet:port_number(),
+          tail_host :: inet:ip_address() | inet:hostname(),
+          tail_port :: inet:port_number()
+         }).
 
 start() ->
-    start(#{ip => "127.0.0.1",
-            port => 5555}).
+    %% maybe start this if it hasn't been
+    application:start(shackle),
+    Cluster = get_cluster(),
+    lager:info("cluster: ~p", [Cluster]),
+    maybe_init_ets(),
+    Chains = start_pools(Cluster),
+    lager:info("chains: ~p", [Chains]),
+    application:set_env(vonnegut, chains, Chains),
+    refresh_topic_map(),
+    lager:info("OK"),
+    ok.
+
+start_pools(Chains) ->
+    [begin
+         lager:info("starting chain: ~p", [Chain]),
+         HeadName = make_pool_name(Chain#chain.name, write),
+         start_pool(HeadName, #{ip => Chain#chain.head_host,
+                            port => Chain#chain.head_port}),
+         TailName = make_pool_name(Chain#chain.name, read),
+         start_pool(TailName, #{ip => Chain#chain.tail_host,
+                            port => Chain#chain.tail_port}),
+         Chain#chain.name
+     end
+     || Chain <- Chains].
+
+refresh_topic_map() ->
```

Comment:
this whole mechanism is sketchy.  also we need something like a "config epoch" to be returned with each request, and if we see that it's been incremented, we should refetch.